contrib/upgrade.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+# this is an upgrade script, for gitea deployed on linux as systemd service
+# depends on: curl, xz, sha256sum, gpg
+# usage:      upgrade.sh [version]
+
+# change the variables below for your local setup
+giteaversion=${1:-1.15.10}
+giteabin=/usr/local/bin/gitea
+giteahome=/var/lib/gitea
+giteaconf=/etc/gitea/app.ini
+giteauser="git"
+giteacmd="sudo -u $giteauser $giteabin -c $giteaconf -w $giteahome"
+
+# download new binary
+binname=gitea-${giteaversion}-linux-amd64
+binurl="https://dl.gitea.io/gitea/${giteaversion}/${binname}.xz"
+echo downloading $binurl
+cd $giteahome # needed for gitea dump later
```

Comment:
Maybe re-order this? Such that the echo is afterwards this `cd`, `cd` could fail because of a non-existent path on the system. Which would be weird when getting a downloading notice before-hand 😅.

Reply:
Also, I think
```suggestion
pushd $(pwd);
cd $giteahome # needed for gitea dump later
```
would be nice, especially with a `popd` at the end.